Source/ACE/Entity/Container.cs

```diff
@@ -62,66 +61,51 @@ public virtual void AddToInventory(WorldObject inventoryItem, uint placement = 0
             ActionChain actionChain = new ActionChain();
             actionChain.AddAction(this, () =>
             {
-                if (inventory.ContainsKey(inventoryItem.Guid))
+                if (Inventory.ContainsKey(inventoryItem.Guid))
                 {
                     // if item exists in the list, we are going to shift everything greater than the moving item down 1 to reflect its removal
-                    inventory.Where(i => i.Value.Placement > (uint)inventory[inventoryItem.Guid].Placement).ToList().ForEach(i => i.Value.Placement--);
-                    inventory.Remove(inventoryItem.Guid);
+                    Inventory.Where(i => i.Value.Placement > (uint)Inventory[inventoryItem.Guid].Placement).ToList().ForEach(i => i.Value.Placement--);
+                    Inventory.Remove(inventoryItem.Guid);
                 }
                 // If not going on the very end (next open slot), make a hole.
-                inventory.Where(i => i.Value.Placement >= placement).ToList().ForEach(i => i.Value.Placement++);
+                Inventory.Where(i => i.Value.Placement >= placement).ToList().ForEach(i => i.Value.Placement++);
                 inventoryItem.Placement = placement;
-                inventory.Add(inventoryItem.Guid, inventoryItem);
-                // FIXME: I know there must be a better way, but not sure how to do it.
-                // For now, sync up underlying aceObject
-                AceObject.Inventory.Clear();
-                inventory.ToList().ForEach(i => AceObject.Inventory.Add(i.Value.SnapShotOfAceObject()));
+                var aceO = inventoryItem.SnapShotOfAceObject();
+                Inventory.Add(inventoryItem.Guid, aceO);
             });
             actionChain.EnqueueChain();
         }
 
         public bool HasItem(ObjectGuid inventoryItemGuid, bool includeSubContainers = true)
         {
             if (!includeSubContainers)
-                return inventory.ContainsKey(inventoryItemGuid);
+                return Inventory.ContainsKey(inventoryItemGuid);
 
-            var containers = inventory.Where(wo => wo.Value.ItemCapacity > 0).ToList();
-            return containers.Any(cnt => ((Container)cnt.Value).inventory.ContainsKey(inventoryItemGuid));
+            var containers = Inventory.Where(wo => wo.Value.ItemsCapacity > 0).ToList();
```

Comment:
That was legacy from when we did not have types set and everything was a debug object, I will fix.      This was a lot more involved than I thought, so I just overlooked that. Thanks for catching it!